api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/admin/SparqlQueryController.java

```diff
@@ -110,11 +112,20 @@ private void respondToQuery(HttpServletRequest req, HttpServletResponse resp)
 				.getRDFService();
 
 		String queryString = req.getParameter("query");
+		boolean download = Boolean.parseBoolean(req.getParameter("download"));
+		Query query = SparqlQueryUtils.create(queryString);
 		try {
-			String format = interpretRequestedFormats(req, queryString);
+			String format = interpretRequestedFormats(req, query);
 			SparqlQueryApiExecutor core = SparqlQueryApiExecutor.instance(
 					rdfService, queryString, format);
 			resp.setContentType(core.getMediaType());
+
+			if (download) {	
+				String extension = getFilenameExtension(req, query, format);
+				resp.setHeader("Content-Transfer-Encoding", "binary");
+				resp.setHeader("Content-disposition", "attachment; filename=sparqlquery." + extension);
```

Comment:
This looks and works well. My only minor suggestion is to potentially change the downloaded filename to "query-results" ... instead of "sparqlquery".

Reply:
Sure, I just went with sparqlquery because that's what the filename was when the browser can't support the file type.